spring-cloud-kubernetes-loadbalancer/src/main/java/org/springframework/cloud/kubernetes/loadbalancer/KubernetesLoadBalancerMode.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.loadbalancer;
+
+/**
+ * @author Piotr Minkowski
+ */
+public enum KubernetesLoadBalancerMode {
+
+	/**
+	 * using pod ip and port.
+	 */
+	POD,
```

Comment:
These enums are currently never used. Kubernetes LoadBalancer Ribbon allowed for load-balancing either using pod endpoints (using the `POD` mode) or services (using the `SERVICE` mode). We would like to maintain this feature parity.

Reply:
The default value is POD. In case of SERVICE mode I'm providing dedicated implementation of `ListSupplier`.
```
	@Bean
	@ConditionalOnProperty(name = "spring.cloud.kubernetes.loadbalancer.mode",
			havingValue = "SERVICE")
	KubernetesServicesListSupplier kubernetesServicesListSupplier(Environment environment,
			KubernetesClient kubernetesClient, KubernetesServiceInstanceMapper mapper,
			KubernetesDiscoveryProperties discoveryProperties) {
		return new KubernetesServicesListSupplier(environment, kubernetesClient, mapper,
				discoveryProperties);
	}
```

Reply:
Ok, so you're getting the endpoint-based implementation by using `KubernetesDiscoveryClient` underneath?

Reply:
Yes.